### ISSUE TITLE
Configurable Partial deletion 

### ISSUE THREAD

**Issue by GelidElf** [jknack/handlebars.java#275]
**tl;dr** Would it be possible to make the behaviour from the hash delete-after-merge=true as default via configuration? 

After a day of figuring out what was happening when some of our templates where misbehaving we discovered the caching and the hash that deletes the partials after merging.

Following is an example of code that by default does not work as intended with this implementation, but does with others.

Block:

``` html
<div "component">
  <div "header">
    {{#block "collapse"}}
      <button class="collapse">Collapse</button>
    {{/block}}
    <h3>{{#block "title"}}{{/block}}</h3>
    {{#block "close"}}
      <button class="close">Close</button>
    {{/block}}
  </div>
  <div class="content">
    {{#block "content"}}{{/block}}
  </div>
  {{#block "footer" delete-after-merge=true}}
    <div class="footer">
      {{#block "footercontent"}}{{/block}}
    </div>
  {{/block}}
</div>
```

Partial Case 1:

``` html
{{#partial "collapse"}}
<!-- without collapse -->
{{/partial}}
{{#partial "title"}}
  Without collapse and footer
{{/partial}}
<!-- with close -->
{{#partial "content"}}
  Content
{{/partial}}
{{#partial "footer"}}
<!-- without footer -->
{{/partial}}
```

Definition case 1:
The intent is to show the layer without the collapse button and no content in the footer section.

Partial Case 2:

``` html
<!-- with collapse  -->
{{#partial "title"}}
  With collapse and footer
{{/partial}}
<!-- with close -->
{{#partial "content"}}
  Content
{{/partial}}
{{#partial "footercontent"}}
  With footer content
{{/partial}}
```

Definition case 2:
For this case we want show the collapse button and the content in the footer.

If we include both cases in the same template. The footer behaves correct in every implementation we use, due to the fact that we have included the hash. The collapse section on the other hand is not overriden. Having to use systematically delete-after-merge=true in every block, although fully valid, includes a lot of boilerplate code that we would like to remove.
